src/Html/Parser.elm

```diff
@@ -267,6 +278,103 @@ element =
+            in
+            Parser.oneOf
+                [ -- HTML comments are, albeit considered a bad practice recently,
+                  -- allowed inside <script> to hide scripts from really ancient web browser
+                  comment
+                    |> Parser.map accumulate
+                , Parser.lineComment "//"
+                    |> Parser.getChompedString
+                    |> Parser.map (Text >> accumulate)
+                , Parser.multiComment "/*" "*/" Parser.NotNestable
+                    |> Parser.getChompedString
+                    |> Parser.map (Text >> accumulate)
+                , javaScriptStringLike '"'
+                    |> Parser.map (Text >> accumulate)
+                , javaScriptStringLike '\''
+                    |> Parser.map (Text >> accumulate)
+                , javaScriptStringLike '`'
+                    |> Parser.map (Text >> accumulate)
+                , closingScriptTag
+                    |> Parser.map (\() -> Parser.Done (List.reverse acc))
+                , Parser.chompIf (always True)
+                    |> Parser.getChompedString
+                    |> Parser.map (Text >> accumulate)
+                ]
+
+
+closingScriptTag : Parser ()
+closingScriptTag =
+    Parser.token "</"
+        |. (Parser.chompWhile (\char -> char /= '>' && not (isSpaceCharacter char))
+                |> Parser.getChompedString
+                |> Parser.andThen
+                    (\chunk ->
+                        if String.toLower chunk == "script" then
+                            Parser.succeed ()
+
+                        else
+                            Parser.problem "not a </script>"
+                    )
+           )
+        |. Parser.chompWhile isSpaceCharacter
+        |. Parser.token ">"
+
+
+javaScriptStringLike : Char -> Parser String
+javaScriptStringLike terminatorChar =
+    let
+        terminatorStr =
+            String.fromChar terminatorChar
+    in
+    Parser.succeed identity
+        |. Parser.token terminatorStr
+        |= Parser.loop "" (stringHelp terminatorChar terminatorStr)
+        -- Restoring original shape
+        |> Parser.map (\chunk -> terminatorStr ++ chunk ++ terminatorStr)
+
+
+stringHelp : Char -> String -> String -> Parser (Parser.Step String String)
+stringHelp terminatorChar terminatorStr acc =
+    Parser.oneOf
+        [ Parser.succeed (\char -> Parser.Loop (acc ++ "\\" ++ char))
+            |. Parser.token "\\"
+            |= justOneChar
+        , Parser.token terminatorStr
+            |> Parser.map (\_ -> Parser.Done acc)
+        , Parser.chompWhile (\char -> char /= '\\' && char /= terminatorChar)
+            |> Parser.getChompedString
+            |> Parser.map (\chunk -> Parser.Loop (acc ++ chunk))
+        ]
+
+
+justOneChar : Parser String
+justOneChar =
+    Parser.loop () <|
+        \_ ->
+            Parser.chompIf (always True)
+                |> Parser.getChompedString
+                |> Parser.map Parser.Done
```

Comment:
This "consuming just one whatever character (after backslash escape)" is somewhat rough, but couldn't come up with better impl